raven/utilities/gis.py

```diff
@@ -191,20 +226,69 @@ def get_raster_wcs(coordinates, geographic=True, layer=None):
         return data
 
 
-def get_hydrobasins_location_wfs(coordinates=None, level=12, lakes=True):
+def hydrobasins_domain(
```

Comment:
The reason for the extract is because the source files are zipped to reduce their size. You are right about the archive_sniffer though, it's designed to unzip a file and find the first True value inside. I can probably remove a lot of code within the function using that itertools implementation.

`archive_sniffer` would be worth refactoring but seeing as it's used everywhere, I don't want to break it right now. I wouldn't mind looking at my old OOP-based approach to see if that could better handle the GIS functions in the new year.

Reply:
I guess my point is that archive sniffer *should not* be used everywhere. I would expect one instance of it in each individual process IO, but it feels off to use it inside `gis.py` functions. It could be used once to open local shapefiles, but that's it. This goes back to an early suggestion I made to choose an internal object that will be used to communicate information between functions. So you have
```outside world -> convert to internal object ->  chain of internal functions -> export outside```
At the moment, I have the feeling that each internal function is designed to interact with the outside world and support many different types of inputs. This creates a lot of maintenance and potential for bugs.